src/bundle/Resources/public/js/scripts/admin.notifications.modal.js

```diff
@@ -1,89 +1,171 @@
+
+            refreshNotificationsPage();
+        }
     };
-    const getNotificationsList = () => {
-        const requestOptions = {
+    const setNotificationPage = (pageHtml) => {
+        const modalResults = modal.querySelector(SELECTOR_MODAL_RESULTS);
+
+        modalResults.innerHTML = pageHtml;
+        toggleLoading(true);
+    };
+    const toggleSpinner = (force) => {
+        const spinner = modal.querySelector(SELECTOR_MODAL_SPINNER);
+
+        spinner.classList.toggle(CLASS_MODAL_SPINNER_INVISIBLE, force);
+    };
+    const toggleResults = (force) => {
+        const results = modal.querySelector(SELECTOR_MODAL_RESULTS);
+
+        results.classList.toggle(CLASS_MODAL_RESULTS_INVISIBLE, force);
+    };
+    const toggleLoading = (force) => {
+        toggleSpinner(force);
+        toggleResults(!force);
+    };
+    const fetchNotificationPage = (link) => {
+        if (!link) {
+            return;
+        }
+
+        const request = new Request(link, {
+            method: 'GET',
             headers: {
-                'X-CSRF-Token': token
+                Accept: 'text/html',
             },
+            credentials: 'same-origin',
             mode: 'cors',
-            credentials: 'same-origin'
-        };
+        });
 
-        fetch(modal.querySelector(SELECTOR_TABLE).dataset.notifications, requestOptions)
-            .then(response => response.text())
-            .then(notifications => doc.querySelector(SELECTOR_TABLE_BODY).innerHTML = notifications);
+        currentPageLink = link;
+        toggleLoading(false);
+        fetch(request)
+            .then(handleRequestResponseText)
+            .then(setNotificationPage)
+            .catch(handleResponseError);
     };
-    const setPendingNotificationCount = (notificationsInfo) => {
-        const methodName = notificationsInfo.pending ? 'add' : 'remove';
-        const userAvatar = doc.querySelector('.ez-user-menu__avatar-wrapper');
+    const refreshNotificationsPage = () => {
+        fetchNotificationPage(currentPageLink);
+    };
+    const handleModalResultsClick = (event) => {
+        handleNotificationsPageChange(event);
+        handleTableClick(event);
+    };
+    const handleNotificationsPageChange = (event) => {
+        const { target } = event;
+        const isPaginationBtn = target.classList.contains(CLASS_PAGINATION_LINK);
 
-        userAvatar.setAttribute('data-count', notificationsInfo.pending);
-        userAvatar.classList[methodName]('n-pending-notifications');
-        doc.querySelector('.ez-user-menu__item--notifications').setAttribute('data-count', `(${notificationsInfo.pending})`);
+        if (!isPaginationBtn) {
+            return;
+        }
 
-        if (notificationsCount !== notificationsInfo.total) {
-            notificationsCount = notificationsInfo.total;
+        const notificationsPageLink = target.href;
 
-            getNotificationsList();
-        }
+        event.preventDefault();
```

Comment:
please use `preventDefault` always on the top of the method.

Reply:
But before I check `if (!isPaginationBtn)` I shoud not handle this. So should I put it over: `const notificationsPageLink = target.href;`?

Reply:
Now I see it better. 
The check `if (!isPaginationBtn)` should not be in this method. The `handleModalResultsClick` should work as a dispatcher, here should be checked what was clicked and called the proper function. Not calling everything and inside the method check if it's correct one.